### ISSUE TITLE
[v1.16.0-rc.0] Istio crashes with memory nil pointer

### ISSUE THREAD

**Issue by SantiMunoz** [istio/istio#41918]
### Bug Description

I'm trying the release v1.16.0-rc.0 in a new cluster, but the istiod pods are eventually failing with the following error:
```
2022-11-10T09:55:42.740844Z	info	ads	LDS: PUSH request for node:istio-ingressgateway-microservices-559db4bf9f-g27gl.istio-system resources:0 size:0B
2022-11-10T09:55:44.320471Z	info	ads	Push debounce stable[5] 1 for config ServiceEntry/kube-system/api-internal.kube-system.svc.cluster.local: 101.069524ms since last change, 101.06933ms since last push, full=true
2022-11-10T09:55:44.323004Z	info	ads	XDS: Pushing:2022-11-10T09:55:44Z/3 Services:31 ConnectedEndpoints:20 Version:2022-11-10T09:55:44Z/3
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x29e7cba]

goroutine 754 [running]:
istio.io/istio/pilot/pkg/model.(*PushContext).ServiceAttachedToGateway(...)
	istio.io/istio/pilot/pkg/model/push_context.go:798
istio.io/istio/pilot/pkg/xds.checkProxyDependencies(0xc001601540, {0x19, {0xc002df2c30, 0x2a}, {0xc002ecaa60, 0xb}}, 0xc0015f4680)
	istio.io/istio/pilot/pkg/xds/proxy_dependencies.go:71 +0xba
istio.io/istio/pilot/pkg/xds.ConfigAffectsProxy(0xc001c147e0, 0xc001601540)
	istio.io/istio/pilot/pkg/xds/proxy_dependencies.go:52 +0x148
istio.io/istio/pilot/pkg/xds.DefaultProxyNeedsPush(0xc001601540, 0xc001c147e0)
	istio.io/istio/pilot/pkg/xds/proxy_dependencies.go:93 +0x30
istio.io/istio/pilot/pkg/xds.(*DiscoveryServer).pushConnection(0xc000ed83c0, 0xc00107b540, 0xc00107b540?)
	istio.io/istio/pilot/pkg/xds/ads.go:744 +0x77
istio.io/istio/pilot/pkg/xds.(*DiscoveryServer).Stream(0xc000ed83c0, {0x39c26d0, 0xc002e07f80})
	istio.io/istio/pilot/pkg/xds/ads.go:340 +0xa70
istio.io/istio/pilot/pkg/xds.(*DiscoveryServer).StreamAggregatedResources(0xc002f79b68?, {0x39c26d0?, 0xc002e07f80?})
	istio.io/istio/pilot/pkg/xds/ads.go:237 +0x25
github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3._AggregatedDiscoveryService_StreamAggregatedResources_Handler({0x3499360?, 0xc000ed83c0}, {0x39be328?, 0xc001bec270})
	github.com/envoyproxy/go-control-plane@v0.10.3-0.20220908185826-2b9739221838/envoy/service/discovery/v3/ads.pb.go:298 +0x9f
google.golang.org/grpc.(*Server).processStreamingRPC(0xc0011394a0, {0x39c53f8, 0xc0035d6d00}, 0xc001bf2a20, 0xc000f862a0, 0x59ba900, 0x0)
	google.golang.org/grpc@v1.49.0/server.go:1564 +0xfc7
google.golang.org/grpc.(*Server).handleStream(0xc0011394a0, {0x39c53f8, 0xc0035d6d00}, 0xc001bf2a20, 0x0)
	google.golang.org/grpc@v1.49.0/server.go:1646 +0x9ea
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.49.0/server.go:938 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.49.0/server.go:936 +0x28a
```

I'm using the following Istio Operator, which works in our cluster running Istio 1.14.1:
<details>
  <summary>Show IstioOperator</summary>
  
  ```
➜ kubectl describe istiooperator istio-controlplane -n istio-system
Name:         istio-controlplane
Namespace:    istio-system
Labels:       app.kubernetes.io/managed-by=Helm
Annotations:  meta.helm.sh/release-name: istio-operator
              meta.helm.sh/release-namespace: istio-system
API Version:  install.istio.io/v1alpha1
Kind:         IstioOperator
Metadata:
<--redacted-->
Spec:
  Components:
    Ingress Gateways:
      Enabled:  true
      k8s:
        Affinity:
          Node Affinity:
            Required During Scheduling Ignored During Execution:
              Node Selector Terms:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
          Pod Anti Affinity:
            Required During Scheduling Ignored During Execution:
              Label Selector:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
              Topology Key:  kubernetes.io/hostname
        Hpa Spec:
          Max Replicas:  5
          Min Replicas:  3
        Overlays:
          API Version:  apps/v1
          Kind:         Deployment
          Name:         istio-ingressgateway-BBB
          Patches:
            Path:  spec.template.spec.topologySpreadConstraints
            Value:
              Label Selector:
                Match Labels:
                  App:             istio-ingressgateway-BBB
              Max Skew:            1
              Topology Key:        topology.kubernetes.io/zone
              When Unsatisfiable:  ScheduleAnyway
        Resources:
          Limits:
            Cpu:     2000m
            Memory:  1Gi
          Requests:
            Cpu:     200m
            Memory:  128Mi
        Service:
          Ports:
            Name:         https
            Port:         443
            Protocol:     TCP
            Target Port:  8443
        Service Annotations:
          service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags:           <--redacted-->
          service.beta.kubernetes.io/aws-load-balancer-backend-protocol:                   ssl
          service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled:  true
          service.beta.kubernetes.io/aws-load-balancer-internal:                           true
          service.beta.kubernetes.io/aws-load-balancer-ssl-cert:                           <--redacted-->
          service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy:             ELBSecurityPolicy-TLS13-1-2-2021-06
          service.beta.kubernetes.io/aws-load-balancer-ssl-ports:                          443
          service.beta.kubernetes.io/aws-load-balancer-target-node-labels:                 n26.com/workload-type=XXX
          service.beta.kubernetes.io/aws-load-balancer-type:                               nlb
        Tolerations:
          Effect:    NoSchedule
          Key:       n26.com/dedicated
          Operator:  Equal
          Value:     XXX
      Label:
        App:    istio-ingressgateway-BBB
        Istio:  ingressgateway-BBB
      Name:     istio-ingressgateway-BBB
      Enabled:  true
      k8s:
        Affinity:
          Node Affinity:
            Required During Scheduling Ignored During Execution:
              Node Selector Terms:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
          Pod Anti Affinity:
            Required During Scheduling Ignored During Execution:
              Label Selector:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
              Topology Key:  kubernetes.io/hostname
        Hpa Spec:
          Max Replicas:  10
          Min Replicas:  3
        Overlays:
          API Version:  apps/v1
          Kind:         Deployment
          Name:         istio-ingressgateway-AAA
          Patches:
            Path:  spec.template.spec.topologySpreadConstraints
            Value:
              Label Selector:
                Match Labels:
                  App:             istio-ingressgateway-AAA
              Max Skew:            1
              Topology Key:        topology.kubernetes.io/zone
              When Unsatisfiable:  ScheduleAnyway
        Resources:
          Limits:
            Cpu:     2000m
            Memory:  1Gi
          Requests:
            Cpu:     200m
            Memory:  128Mi
        Service:
          Ports:
            Name:         https
            Port:         443
            Protocol:     TCP
            Target Port:  8443
        Service Annotations:
          service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags:           <--redacted-->
          service.beta.kubernetes.io/aws-load-balancer-backend-protocol:                   ssl
          service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled:  true
          service.beta.kubernetes.io/aws-load-balancer-internal:                           true
          service.beta.kubernetes.io/aws-load-balancer-ssl-ports:                          443
          service.beta.kubernetes.io/aws-load-balancer-target-node-labels:                 n26.com/workload-type=XXX
          service.beta.kubernetes.io/aws-load-balancer-type:                               nlb
        Tolerations:
          Effect:    NoSchedule
          Key:       n26.com/dedicated
          Operator:  Equal
          Value:     XXX
      Label:
        App:    istio-ingressgateway-AAA
        Istio:  ingressgateway-AAA
      Name:     istio-ingressgateway-AAA
      Enabled:  true
      k8s:
        Affinity:
          Node Affinity:
            Required During Scheduling Ignored During Execution:
              Node Selector Terms:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
          Pod Anti Affinity:
            Required During Scheduling Ignored During Execution:
              Label Selector:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
              Topology Key:  kubernetes.io/hostname
        Hpa Spec:
          Max Replicas:  5
          Min Replicas:  3
        Resources:
          Limits:
            Cpu:     2000m
            Memory:  1Gi
          Requests:
            Cpu:     200m
            Memory:  128Mi
        Service:
          Ports:
            Name:         https
            Port:         443
            Protocol:     TCP
            Target Port:  8443
        Service Annotations:
          service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags:          <--redacted-->
          service.beta.kubernetes.io/aws-load-balancer-backend-protocol:                   tcp
          service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled:  true
          service.beta.kubernetes.io/aws-load-balancer-extra-security-groups:              <--redacted-->
          service.beta.kubernetes.io/aws-load-balancer-internal:                           false
          service.beta.kubernetes.io/aws-load-balancer-target-node-labels:                 n26.com/workload-type=XXX
        Tolerations:
          Effect:    NoSchedule
          Key:       n26.com/dedicated
          Operator:  Equal
          Value:     XXX
      Label:
        App:    istio-ingressgateway-CCC
        Istio:  ingressgateway-CCC
      Name:     istio-ingressgateway-CCC
      Enabled:  true
      k8s:
        Affinity:
          Node Affinity:
            Required During Scheduling Ignored During Execution:
              Node Selector Terms:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
          Pod Anti Affinity:
            Required During Scheduling Ignored During Execution:
              Label Selector:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    XXX
              Topology Key:  kubernetes.io/hostname
        Env:
          Name:   ISTIO_META_REQUESTED_NETWORK_VIEW
          Value:  <--redacted-->-1668013954518388309
        Hpa Spec:
          Max Replicas:  5
          Min Replicas:  3
        Overlays:
          API Version:  apps/v1
          Kind:         Deployment
          Name:         istio-eastwestgateway
          Patches:
            Path:  spec.template.spec.topologySpreadConstraints
            Value:
              Label Selector:
                Match Labels:
                  App:             istio-eastwestgateway
              Max Skew:            1
              Topology Key:        topology.kubernetes.io/zone
              When Unsatisfiable:  ScheduleAnyway
        Service:
          Ports:
            Name:         status-port
            Port:         15021
            Target Port:  15021
            Name:         tls
            Port:         15443
            Target Port:  15443
            Name:         tls-istiod
            Port:         15012
            Target Port:  15012
            Name:         tls-webhook
            Port:         15017
            Target Port:  15017
        Service Annotations:
          service.beta.kubernetes.io/aws-load-balancer-additional-resource-tags:
          service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled:  true
          service.beta.kubernetes.io/aws-load-balancer-internal:                           true
          service.beta.kubernetes.io/aws-load-balancer-target-node-labels:                 n26.com/workload-type=XXX
          service.beta.kubernetes.io/aws-load-balancer-type:                               nlb
        Tolerations:
          Effect:    NoSchedule
          Key:       n26.com/dedicated
          Operator:  Equal
          Value:     XXX
      Label:
        App:                        istio-eastwestgateway
        Istio:                      eastwestgateway
        topology.istio.io/network:  <--redacted-->-1668013954518388309
      Name:                         istio-eastwestgateway
    Pilot:
      k8s:
        Affinity:
          Node Affinity:
            Required During Scheduling Ignored During Execution:
              Node Selector Terms:
                Match Expressions:
                  Key:       n26.com/workload-type
                  Operator:  In
                  Values:
                    core
          Pod Anti Affinity:
            Required During Scheduling Ignored During Execution:
              Label Selector:
                Match Expressions:
                  Key:       app
                  Operator:  In
                  Values:
                    istiod
              Topology Key:  kubernetes.io/hostname
              Label Selector:
                Match Expressions:
                  Key:       app.kubernetes.io/instance
                  Operator:  In
                  Values:
                    core
              Topology Key:  kubernetes.io/hostname
        Env:
          Name:   RESOLVE_HOSTNAME_GATEWAYS
          Value:  true
          Name:   ENABLE_CA_SERVER
          Value:  false
          Name:   ENABLE_DEBUG_ON_HTTP
          Value:  false
          Name:   PILOT_FILTER_GATEWAY_CLUSTER_CONFIG
          Value:  true
        Hpa Spec:
          Behavior:
            Scale Down:
              Stabilization Window Seconds:  3600
          Max Replicas:                      10
          Min Replicas:                      3
        Overlays:
          API Version:  apps/v1
          Kind:         Deployment
          Name:         istiod
          Patches:
            Path:   spec.template.spec.containers.[name:discovery].args[-1]
            Value:  --grpcAddr=
            Path:   spec.template.spec.containers.[name:discovery].args[-1]
            Value:  --tlsCertFile=/etc/cert-manager/tls/tls.crt
            Path:   spec.template.spec.containers.[name:discovery].args[-1]
            Value:  --tlsKeyFile=/etc/cert-manager/tls/tls.key
            Path:   spec.template.spec.containers.[name:discovery].args[-1]
            Value:  --caCertFile=/etc/cert-manager/ca/root-cert.pem
            Path:   spec.template.spec.containers.[name:discovery].volumeMounts[-1]
            Value:
              Mount Path:  /etc/cert-manager/tls
              Name:        cert-manager
              Read Only:   true
            Path:          spec.template.spec.containers.[name:discovery].volumeMounts[-1]
            Value:
              Mount Path:  /etc/cert-manager/ca
              Name:        ca-root-cert
              Read Only:   true
            Path:          spec.template.spec.volumes[-1]
            Value:
              Name:  cert-manager
              Secret:
                Secret Name:  istiod-tls
            Path:             spec.template.spec.volumes[-1]
            Value:
              Config Map:
                Default Mode:  420
                Name:          istio-ca-root-cert
              Name:            ca-root-cert
            Path:              spec.template.spec.topologySpreadConstraints
            Value:
              Label Selector:
                Match Labels:
                  App:             istiod
              Max Skew:            1
              Topology Key:        topology.kubernetes.io/zone
              When Unsatisfiable:  ScheduleAnyway
        Resources:
          Requests:
            Cpu:     30m
            Memory:  800M
        Service Annotations:
          ad.datadoghq.com/endpoints.check_names:   ["istio"]
          ad.datadoghq.com/endpoints.init_configs:  [{}]
          ad.datadoghq.com/endpoints.instances:     [
  {
    "istiod_endpoint": "http://%%host%%:15014/metrics",
    "send_histograms_buckets": true,
    "use_openmetrics": true
  }
]

        Tolerations:
          Effect:    NoSchedule
          Key:       n26.com/dedicated
          Operator:  Equal
          Value:     core
  Hub:               artifactory.cd-tech26.de/docker/istio
  Mesh Config:
    Access Log File:  /dev/stdout
    Default Config:
      Hold Application Until Proxy Starts:  true
      Proxy Metadata:
        ISTIO_META_DNS_AUTO_ALLOCATE:  true
        ISTIO_META_DNS_CAPTURE:        true
    Discovery Selectors:
      Match Expressions:
        Key:       name
        Operator:  NotIn
        Values:
          <--redacted-->
    Extension Providers:
      Envoy Ext Authz Http:
        Headers To Downstream On Deny:
          content-type
          set-cookie
        Headers To Upstream On Allow:
          authorization
          cookie
          path
        Include Additional Headers In Check:
          X - Auth - Request - Redirect:  %REQ(x-forwarded-proto)%://%REQ(:authority)%%REQ(:path)%
        Include Headers In Check:
          authorization
          cookie
        Port:     80
        Service:  oauth2-proxy.core-CCC.svc.cluster.local
      Name:       oauth2-proxy
    Mesh MTLS:
      Min Protocol Version:  TLSV1_3
  Profile:                   minimal
  Values:
    Global:
      Ca Address:  cert-manager-istio-csr.cert-manager.svc:443
      Mesh ID:     istio-mesh
      Multi Cluster:
        Cluster Name:  <--redacted-->-1668013954518388309
      Network:         <--redacted-->-1668013954518388309
      Proxy:
        Cluster Domain:  cluster.local
        Image:           proxyv2
        Log Level:       warning
        Resources:
          Limits:
            Cpu:     2000m
            Memory:  1024Mi
          Requests:
            Cpu:     100m
            Memory:  128Mi
        Tracer:      datadog
    Pilot:
      Trace Sampling:  100
Status:
  Component Status:
    Base:
      Status:  HEALTHY
    Ingress Gateways:
      Status:  HEALTHY
    Pilot:
      Error:  failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
  Deployment/istio-system/istiod (container failed to start: CrashLoopBackOff: back-off 20s restarting failed container=discovery pod=istiod-58d4759ff5-zqf64_istio-system(a1f1ee7f-205c-449c-ab1a-5d54605b4298))
      Status:  ERROR
  Status:      ERROR
Events:        <none>
  ```
</details>

### Version

```prose
✗ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.2
Kustomize Version: v4.5.4
Server Version: v1.23.13

I can't run istioctl because istiod is down
```

**Comment by chaunceyjiang** [comment]
I want to /assign, I am a newbie.I will try to find the cause of this problem. I'm not sure I can fix it

**Comment by SantiMunoz** [comment]
Thanks @chaunceyjiang for taking care of it 🙇 
Do you know how can we avoid this issue? is there any workaround while we wait for the fix to be released?

**Comment by dddddai** [comment]
Sorry it was my bad..

> is there any workaround while we wait for the fix to be released?

I think you enabled `PILOT_FILTER_GATEWAY_CLUSTER_CONFIG`? You can disable it as a temporary workaround

**Comment by SantiMunoz** [comment]
Thanks for the insights @dddddai, I see the fix will be part of the 1.16 final release isn't it? otherwise we will disable `PILOT_FILTER_GATEWAY_CLUSTER_CONFIG` temporary

**Comment by dddddai** [comment]
> I see the fix will be part of the 1.16.0 final release isn't it?

Yes the fix has been cherry picked to 1.16 branch and should be shipped in 1.16 release
